server/eclipse-project/src/main/java/us/freeandfair/corla/csv/ColoradoBallotManifestParser.java

```diff
@@ -117,20 +119,27 @@ public ColoradoBallotManifestParser(final String the_string, final Instant the_t
    */
   private BallotManifestInfo extractBMI(final CSVRecord the_line,
                                         final Instant the_timestamp) {
+    BallotManifestInfo result = null;
+    
     try {
-      return Persistence.get(new BallotManifestInfo(the_timestamp, 
-                                                    the_line.get(COUNTY_ID_COLUMN),
-                                                    the_line.get(SCANNER_ID_COLUMN),
-                                                    the_line.get(BATCH_NUMBER_COLUMN),
-                                                    Integer.parseInt(the_line.
-                                                                     get(NUM_BALLOTS_COLUMN)),
-                                                    the_line.get(BATCH_LOCATION_COLUMN)),
-                             BallotManifestInfo.class);
-    } catch (final NumberFormatException e) {
-      return null;
-    } catch (final ArrayIndexOutOfBoundsException e) {
-      return null;
+      // obtain the county from the ID or name in the line
+      final County c = CountyQueries.fromString(the_line.get(COUNTY_ID_COLUMN));
+      if (c != null) {
+        result = 
+            Persistence.get(new BallotManifestInfo(the_timestamp, 
+                                                   c.identifier(),
+                                                   the_line.get(SCANNER_ID_COLUMN),
```

Comment:
This is not about COUNTY_ID_COLUMN being null, it's about not finding a matching a county in the database. The others can't return null - the worst they'd do is say "there's no column with that number" (ArrayIndexOutOfBoundsException). Of course if the data in the file is bad, that's a problem, but there's no way for us to tell other than the number of ballots having to be numeric.